src/raster/rasterband.rs

```diff
@@ -950,4 +1153,41 @@ impl<'a> ColorTable<'a> {
             a: color_entry.c4,
         })
     }
+
+    /// Set entry in the RasterBand color table.
+    ///
+    /// The `entry` variant type must match `palette_interpretation`.
+    ///
+    /// The table is grown as needed to hold the supplied index, filling in gaps with
+    /// the default [`GDALColorEntry`] value.
+    pub fn set_color_entry(&mut self, index: u16, entry: &ColorEntry) {
+        unsafe { GDALSetColorEntry(self.c_color_table, index as c_int, &entry.into()) }
+    }
+}
+
+impl Drop for ColorTable<'_> {
+    fn drop(&mut self) {
+        if self.rust_owned {
+            unsafe { GDALDestroyColorTable(self.c_color_table) }
+        }
+    }
+}
+
+impl Default for ColorTable<'_> {
+    fn default() -> Self {
+        Self::new(PaletteInterpretation::Rgba)
+    }
+}
+
+impl Debug for ColorTable<'_> {
+    fn fmt(&self, f: &mut Formatter<'_>) -> std::fmt::Result {
+        let entries = (0..self.entry_count())
+            .filter_map(|i| self.entry(i))
+            .collect::<Vec<_>>();
+
+        f.debug_struct("ColorTable")
```

Comment:
Nit-picking again, but I wonder if we shouldn't abbreviate empty (`(0, 0, 0, 0)`) entries at the end or only display the set ones (`[0] => (1, 1, 1, 0), [1] => (2, 2, 2, 0) [...]`).

Totally fine to leave this for another PR though.

Reply:
I'll look into that. I'm kinda annoyed in a `gdalinfo` you end up with a long list of zero entries. Before writing/reading the test file, the table is short. At one time I did some work on the Geotrellis GeoTIFF parser, and seem to remember there's legal to have only the entries you specify. Perhaps this is GDAL GTiff implementation thing.

Reply:
@lnicola I poked at it a bit an think I'll go with saving it for another PR.